p2p/host.go

```diff
@@ -275,30 +244,27 @@ func (h *host) peerCount() int {
 	return len(peers)
 }
 
-// fullAddr returns the first full multiaddress of the host
-func (h *host) fullAddr() (maddrs ma.Multiaddr) {
-	return h.fullAddrs()[0]
+// addrInfos returns the libp2p AddrInfos of the host
+func (h *host) addrInfos() (addrInfos []*peer.AddrInfo, err error) {
+	for _, multiaddr := range h.multiaddrs() {
+		addrInfo, err := peer.AddrInfoFromP2pAddr(multiaddr)
+		if err != nil {
+			return nil, err
+		}
+		addrInfos = append(addrInfos, addrInfo)
+	}
+	return addrInfos, nil
 }
 
-// fullAddrs returns the full multiaddresses of the host
-func (h *host) fullAddrs() (maddrs []ma.Multiaddr) {
+// multiaddrs returns the multiaddresses of the host
+func (h *host) multiaddrs() (multiaddrs []ma.Multiaddr) {
 	addrs := h.h.Addrs()
-	for _, a := range addrs {
-		maddr, err := ma.NewMultiaddr(fmt.Sprintf("%s/p2p/%s", a, h.h.ID()))
+	for _, addr := range addrs {
+		multiaddr, err := ma.NewMultiaddr(fmt.Sprintf("%s/p2p/%s", addr, h.id()))
 		if err != nil {
 			continue
 		}
-		maddrs = append(maddrs, maddr)
-	}
-	return maddrs
-}
-
-// addrInfo returns the libp2p AddrInfo of the host
-func (h *host) addrInfo() (addrInfo *peer.AddrInfo, err error) {
```

Comment:
we don't need this anymore?

Reply:
We are getting `AddrInfo` for each address using `addrInfos()`. The only place the single address was being used was in the tests. which now uses `addrInfo[0]` (the first `AddrInfo`).